### PR TITLE
refactor persistence to use promises

### PR DESCRIPTION
**Pull request by chadwpry**
switch all datastore methods to return a promise and no longer take success and error callbacks

### CODE REVIEW

**Comment by chadwpry** [comment]
@robskrob can you have a look at this PR?

**Comment by robskrob** [comment]
@chadwpry yep -- looks good! 👍

File: extension/background.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-    console.log('background.js', 'send message');
+let jQuery = require('jquery');
+let capture = require('./lib/capture').default;
 
-    chrome.tabs.sendMessage(tabs[0].id, {
-      "type": "capture_product"
-    });
-  });
-});
+//chrome.browserAction.onClicked.addListener((tab) => {
```
Comment (robskrob) [inline extension/background.js:3]: Are you just saving this for later?

File: extension/background.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
-chrome.browserAction.onClicked.addListener(function(tab) {
-  chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-    console.log('background.js', 'send message');
+let jQuery = require('jquery');
```
Comment (robskrob) [inline extension/background.js]: are we using jQuery?

File: extension/background.js (hunk 1 above)
Comment (chadwpry) [reply extension/background.js:3]: Yes, it is easier to remember how to use it than finding a good example online.

File: extension/content.js [hunk 3]
```diff
@@ -1,18 +1,12 @@
-let capture = require('./lib/capture.js').default;
+let capture = require('./lib/capture').default;
```
Comment (robskrob) [inline extension/content.js:1]: this is beautiful. Nice work!

It looks like we've reversed the paradigm. When the tab loads, we suck the data from the page using constructSample and then send a message to the background script, passing the sample data in the request. The background scripts receives the message and the data from the page and posts to the api. The methods in the content and background scripts are neatly defined in lib/capture.

File: extension/lib/capture.js [hunk 4]
```diff
@@ -45,10 +37,46 @@ let constructProduct = (host, callback) => {
...
+let publishSamples = (object) => {
+  let host = object.host;
+  delete object.host;
+
+  let promise = postJsonPromise('https://localhost/api/v1/samples', JSON.stringify({
+    host: host,
+    payload: object
+  }));
+
+  promise.done((json, state, xhr) => {
```
Comment (robskrob) [inline extension/lib/capture.js:63]: ok so when the request succeeds, we essentially do nothing.

File: models/datastore.js [hunk 5]
```diff
@@ -6,95 +6,70 @@ let client = datastore({
   keyFilename: config.googleCloudKey
```
Comment (robskrob) [inline models/datastore.js:6]: awesome!

**Comment by robskrob** [comment]
👍

File: extension/lib/capture.js (hunk 4 above)
Comment (chadwpry) [reply extension/lib/capture.js:63]: Yes, at least for now.

**Pull request merged**